datafusion/physical-plan/src/aggregates/order/mod.rs

```diff
@@ -18,13 +18,12 @@
 use arrow_array::ArrayRef;
 use arrow_schema::Schema;
 use datafusion_common::Result;
-use datafusion_physical_expr::EmitTo;
-
-use super::{AggregationOrdering, GroupByOrderMode};
+use datafusion_physical_expr::{EmitTo, PhysicalSortExpr};
 
 mod full;
 mod partial;
 
+use crate::windows::PartitionSearchMode;
```

Comment:
This change effectively removes the duplication between `GroupByOrderMode` and `PartitionSearchMode` that represent the same thing right?

I find `PartitionSearchMode` a confusing name as the term "partition" is pretty overloaded already (like each ExecutionPlan has input/output partitions, and WindowExec deals with partitions). 

Also the fact it is in the `windows` module seems to be a mismatch given it is now used in the Aggregation logic

Maybe it could go into its own module 🤔 and be called `SortOrderMode` or something

Also, can we clarify what `PartitionSearchMode::PartiallySorted` means -- specifically, does it represent a prefix of the columns that is sorted?

This could all be done as follow on PRs